lib/Doctrine/Migrations/Tools/Console/ConsoleRunner.php

```diff
@@ -121,4 +128,34 @@ public static function addCommands(Application $cli, ?DependencyFactory $depende
 
         $cli->add(new DiffCommand($dependencyFactory));
     }
+
+    /**
+     * @param mixed $dependencyFactory
+     *
+     * @return mixed
+     */
+    private static function checkLegacyConfiguration($dependencyFactory, string $configurationFile)
+    {
+        if ($dependencyFactory instanceof HelperSet) {
+            $configurations = new ConfigurationFileWithFallback();
+            if ($dependencyFactory->has('em') && $dependencyFactory->get('em') instanceof EntityManagerHelper) {
+                $dependencyFactory = DependencyFactory::fromEntityManager(
+                    $configurations,
+                    new ExistingEntityManager($dependencyFactory->get('em')->getEntityManager())
+                );
+            } elseif ($dependencyFactory->has('db') && $dependencyFactory->get('db') instanceof ConnectionHelper) {
+                $dependencyFactory = DependencyFactory::fromConnection(
+                    $configurations,
+                    new ExistingConnection($dependencyFactory->get('db')->getConnection())
+                );
+            } else {
+                throw new RuntimeException(sprintf(
+                    'Configuration HelperSet returned by "%s" does not have neither the "em" or the "db" helper.',
+                    $configurationFile
+                ));
+            }
+        }
+
+        return $dependencyFactory;
```

Comment:
Using early returns should make this less complex and easier to analyze since we don't reaffect `$dependencyFactory`:

```suggestion
        if (!$dependencyFactory instanceof HelperSet) {
            return $dependencyFactory;
        }
        
        $configurations = new ConfigurationFileWithFallback();
        if ($dependencyFactory->has('em') && $dependencyFactory->get('em') instanceof EntityManagerHelper) {
            return DependencyFactory::fromEntityManager(
                $configurations,
                new ExistingEntityManager($dependencyFactory->get('em')->getEntityManager())
            );
        }
        
        if ($dependencyFactory->has('db') && $dependencyFactory->get('db') instanceof ConnectionHelper) {
            return DependencyFactory::fromConnection(
                $configurations,
                new ExistingConnection($dependencyFactory->get('db')->getConnection())
            );
        }
        throw new RuntimeException(sprintf(
            'Configuration HelperSet returned by "%s" does not have neither the "em" or the "db" helper.',
            $configurationFile
        ));     
```